src/CEOFESABundle/Controller/StructureController.php

```diff
@@ -239,4 +242,32 @@ private function createDeleteForm($id)
             ->getForm()
         ;
     }
+
+    /**
+     * @param int $id
+     *
+     * @Route("/editpdf/{id}", name="structure_pdf")
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function printPartnerCovenantPdfAction($id)
```

Comment:
Use an implicit Doctrine ParamConverter, 
modify your route and method argument to have your structure entity as method argument :)